pkg/gktest/assertion.go

```diff
@@ -81,6 +85,10 @@ func (a *Assertion) matchesCount(matching int32) error {
 func (a *Assertion) matchesCountInt(matching int32) error {
 	wantMatching := a.Violations.IntVal
 	if wantMatching != matching {
+		if a.Message != nil {
+			return fmt.Errorf("%w: got %d violations containing %q but want exactly %d",
```

Comment:
s/containing/matching ?

Reply:
ditto below

Reply:
It's "containing" since the violations are the subject. The messages "match" the violations, and the violations "contain" the messages.

You get a similar pattern for regular expressions. You would say `.*` matches `foo`, but you would not say `foo` matches `.*`. You would say `foo` is matched by `.*`. "Matching" is transitive ("foo" matches "foobar" and "foobar" matches "foobarqux" implies "foo" matches "foobarqux") but not reflexive ("foo" matches "foobar" does not mean "foobar" matches "foo").